central/vulnerabilityrequest/manager/requestmgr/manager_impl.go

```diff
@@ -206,11 +208,45 @@ func (m *managerImpl) Undo(ctx context.Context, id string, _ *common.VulnRequest
 	return req, nil
 }
 
+func (m *managerImpl) Cancel(ctx context.Context, id string) (*storage.VulnerabilityRequest, error) {
+	if !features.UnifiedCVEDeferral.Enabled() {
+		return nil, errors.Errorf("set the environment variable %s=true and retry", features.UnifiedCVEDeferral.EnvVar())
+	}
+	if id == "" {
+		return nil, errox.InvalidArgs.CausedBy("ID must be specified")
+	}
+
+	comment := "[System Generated] Vulnerability exception canceled"
+	user := authn.UserFromContext(ctx)
+	if user == nil {
+		log.Errorf("failed to determine user for vulnerability exception %s", id)
+	} else {
+		comment = fmt.Sprintf("%s by %s", comment, user.GetName())
+	}
+
+	req, err := m.vulnReqs.MarkRequestInactive(ctx, id, comment)
+	if err != nil {
+		return nil, errors.Wrapf(err, "canceling vulnerability request %s", id)
+	}
+	// Revert the exception enforcement.
+	if err := m.UnSnoozeVulnerabilityOnRequest(ctx, req); err != nil {
+		return nil, errors.Wrapf(err, "removing vulnerability request %s enforcement", id)
+	}
+	return req, nil
+}
+
 func (m *managerImpl) UpdateExpiry(ctx context.Context, id string, reqParams *common.VulnRequestParams) (*storage.VulnerabilityRequest, error) {
-	// Currently, only deferral requests can be updated
+	if id == "" {
+		return nil, errox.InvalidArgs.CausedBy("vulnerability exception ID must be specified")
+	}
+	if reqParams == nil {
+		return nil, nil
```

Comment:
This seems like an odd case.  Should it return an invalid args error too?

Reply:
good catch